jenkins-integration/jenkins-jobs/scenarios/couch-environment-timeout/Jenkinsfile

```diff
@@ -0,0 +1,97 @@
+     code_deploy: [
+             type: "r10k",
+             control_repo: "git@github.com:puppetlabs/puppetlabs-puppetserver_perf_control.git",
+             basedir: "/etc/puppetlabs/code-staging/environments",
+             environments: ["production"],
+             hiera_config_source_file: "/etc/puppetlabs/code-staging/environments/production/root_files/hiera.yaml"
+     ],
+     server_java_args: "-Xms12g -Xmx12g",
+     puppet_settings: [
+             main: [
+                     "environment_timeout": "unlimited"
+             ],
+             master: [
+                     "environment_timeout": "unlimited"
+             ]
+     ],
+     background_scripts: [
+             "./jenkins-jobs/common/scripts/background/curl-server-metrics-loop.sh"
+     ],
+     archive_sut_files: [
+             "/var/log/puppetlabs/puppetserver/metrics.json"
+     ]
+    ]
+}
+
+environment_timeout_0_500 = environment_timeout_0_base_config()
+environment_timeout_0_500.put('job_name', 'pe-couch-environment-timeout-0-500')
+environment_timeout_0_500.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-500-2-hours.json')
+
+environment_timeout_unlimited_500 = environment_timeout_unlimited_base_config()
+environment_timeout_unlimited_500.put('job_name', 'pe-couch-environment-timeout-unlimited-500')
+environment_timeout_unlimited_500.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-500-2-hours.json')
+
+environment_timeout_0_1000 = environment_timeout_0_base_config()
+environment_timeout_0_1000.put('job_name', 'pe-couch-environment-timeout-0-1000')
+environment_timeout_0_1000.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1000-2-hours.json')
+
+environment_timeout_unlimited_1000 = environment_timeout_unlimited_base_config()
+environment_timeout_unlimited_1000.put('job_name', 'pe-couch-environment-timeout-unlimited-1000')
+environment_timeout_unlimited_1000.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1000-2-hours.json')
+
+environment_timeout_0_1250 = environment_timeout_0_base_config()
+environment_timeout_0_1250.put('job_name', 'pe-couch-environment-timeout-0-1250')
+environment_timeout_0_1250.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1250-2-hours.json')
+
+environment_timeout_unlimited_1250 = environment_timeout_unlimited_base_config()
+environment_timeout_unlimited_1250.put('job_name', 'pe-couch-environment-timeout-unlimited-1250')
+environment_timeout_unlimited_1250.put('gatling_simulation_config', '../simulation-runner/config/scenarios/pe-couch-medium-1250-2-hours.json')
+
+pipeline.multipass_pipeline([
+        environment_timeout_0_500,
+        environment_timeout_unlimited_500,
+        environment_timeout_0_1000,
+        environment_timeout_unlimited_1000,
+        environment_timeout_0_1250,
+        environment_timeout_unlimited_1250,
+])
```

Comment:
Was trying to figure out a way to shorten this file up a bit, but not sure it's worth it
something like 

```groovy
jobs = []
["500", "1000", "1250].each { -> numAgents
    ["0", "unlimited"].each { -> timeout
        // Put those variables into a copy of the base config
        jobs.add(customizedJob)
    }
}
```

You'd only need one base config, but it'd be harder to change things later if it didn't fit the pattern I guess. I don't really care, wdyt?